### PR TITLE
Adds lockable output buses

### PR DESCRIPTION
**Pull request by querns**
## NOTE: Not yet supported for ME Output Buses!

Implementing this for ME Output Buses turned out to be way harder than I originally thought and would have necessitated a VoidProtectionHelper rewrite, so I'm punting for now. Just as well, given the active development on the ME Output Bus. Would rather get this out there now than get mired in a rewrite of one of the mod's most important pieces of logic.

# Lockable Output Buses

This PR adds locking functionality to output buses and super bus (O) blocks. Either drag an item from NEI to set the lock, or use an item in your inventory. The filter slot is a phantom item slot; no item is actually stored here.

When locked, output buses can only ever receive the filtered item when the multiblock processes recipes. This functionality should reduce the amount of item logistics boilerplate needed for creating automation when more than one item is output by a multiblock machine.

A quick video showing them in action:

https://github.com/user-attachments/assets/17e94a0e-21cf-40c2-a6d3-7b0eea2247f3

The logic for which output bus gets what items is identical to locked output hatches; locked output buses preferentially receive the item in question as long as they have space, then after they fill the output can go to an unlocked output bus if there's room left in it.

Data stick support is also included. Left click while holding a data stick to save the filter configuration to the data stick, then right click to load the configuration. An output bus with an _absence_ of a set filter item will also be saved; loading with such a configured data stick will clear the filter of the output bus you've chosen.

## More Images

java_soX8W8VxuR
java_oCkrKVpBrq
java_ZQqGClJxMr
java_3bhIzPch97

### CODE REVIEW

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java [hunk 1]
```diff
@@ -1302,6 +1302,21 @@ public boolean addOutput(ItemStack aStack) {
         return outputSuccess;
     }
 
+    private boolean dumpItem(List<GT_MetaTileEntity_Hatch_OutputBus> outputBuses, ItemStack itemStack,
+        boolean restrictiveBusesOnly) {
+        for (GT_MetaTileEntity_Hatch_OutputBus outputBus : filterValidMTEs(outputBuses)) {
```
Comment (Nxer) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java:1307]: Move the step `filterValidMTEs` into upper level calling method `addOutput`, then there can decrease a time of loop which is extra and not necessary.

Comment (querns) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java]: should be fixed now

**Review by Nxer — APPROVED** [review]
After testing the whole thing, no abnormalities were found. All functions are running normally, including the Void Protection part.

Well done.

This new feature will greatly improve the convenience of production line construction. 

A more suggestion, add method that let player `Right Click` the filter item stack to cancel the filter setting.

**Comment by querns** [comment]
You can clear the filter with left click. Right click has a special function with this UI element. It lets you restore the filtered item if you accidentally clear it. (I'm just reusing an existing UI element for the filter.)

**Comment by Nxer** [comment]
> Right click has a special function with this UI element. It lets you restore the filtered item if you accidentally clear it. (I'm just reusing an existing UI element for the filter.)

Good job. Bless you.

**Pull request merged**